test/e2e/usage_test.go

```diff
@@ -0,0 +1,209 @@
+	"sigs.k8s.io/e2e-framework/third_party/helm"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource/unstructured/composed"
+
+	apiextensionsv1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+	pkgv1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	"github.com/crossplane/crossplane/test/e2e/config"
+	"github.com/crossplane/crossplane/test/e2e/funcs"
+)
+
+const (
+	// SuiteUsage is the value for the config.LabelTestSuite label to be
+	// assigned to tests that should be part of the Usage test suite.
+	SuiteUsage = "usage"
+)
+
+func init() {
+	environment.AddTestSuite(SuiteUsage,
+		config.WithHelmInstallOpts(
+			helm.WithArgs("--set args={--debug,--enable-usages}"),
+		),
+		config.WithLabelsToSelect(features.Labels{
+			config.LabelTestSuite: []string{SuiteUsage, config.TestSuiteDefault},
+		}),
+	)
+}
+
+// TestUsageStandalone tests scenarios for Crossplane's `Usage` resource without
+// a composition involved.
+func TestUsageStandalone(t *testing.T) {
+	manifests := "test/e2e/manifests/apiextensions/usage/standalone"
+
+	cases := features.Table{
+		{
+			// Deletion of a (used) resource should be blocked if there is a Usage relation with a using resource defined.
+			Name: "UsageBlockedByUsingResource",
+			Assessment: funcs.AllOf(
+				// Create using and used managed resources together with a usage.
+				funcs.ApplyResources(FieldManager, manifests, "with-by/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "with-by/*.yaml"),
+				funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "with-by/usage.yaml", xpv1.Available()),
+
+				// Deletion of used resource should be blocked by usage.
+				funcs.DeletionBlockedByUsageWebhook(manifests, "with-by/used.yaml"),
+
+				// Deletion of using resource should clear usage.
+				funcs.DeleteResources(manifests, "with-by/using.yaml"),
+				funcs.ResourcesDeletedWithin(30*time.Second, manifests, "with-by/using.yaml"),
+				funcs.ResourcesDeletedWithin(30*time.Second, manifests, "with-by/usage.yaml"),
+
+				// Deletion of used resource should be allowed after usage is cleared.
+				funcs.DeleteResources(manifests, "with-by/used.yaml"),
+				funcs.ResourcesDeletedWithin(30*time.Second, manifests, "with-by/used.yaml"),
+			),
+		},
+		{
+			// Deletion of a (protected) resource should be blocked if there is a Usage with a reason defined.
+			Name: "UsageBlockedWithReason",
+			Assessment: funcs.AllOf(
+				// Create protected managed resources together with a usage.
+				funcs.ApplyResources(FieldManager, manifests, "with-reason/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "with-reason/*.yaml"),
+				funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "with-reason/usage.yaml", xpv1.Available()),
+
+				// Deletion of protected resource should be blocked by usage.
+				funcs.DeletionBlockedByUsageWebhook(manifests, "with-reason/used.yaml"),
```

Comment:
we should also assert the reason sent back to users is the one we expect to be.

Reply:
We already have assertions for that at unit tests level. 
Here it should be sufficient to verify that the request is denied by usage webhook.